app/helpers/form_helper.rb

```diff
@@ -91,6 +93,64 @@ def multiple_selects(f, attr, associations, selected_ids, options = {}, html_opt
     end
   end
 
+  def dual_list(f, attr, associations, selected_ids, options = {}, html_options = {})
+    options[:size] = "col-md-10"
+
+    case attr
+    when :organizations
+      klass = Organization
+    when :locations
+      klass = Location
+    else
+      klass = nil
+    end
+
+    authorized = AssociationAuthorizer.authorized_associations(associations.reorder(nil), klass).all
+    unauthorized = selected_ids.blank? ? [] : selected_ids - authorized.map(&:id)
+    items = authorized.map { |item| { value: item.id, label: item.try(:object_label_method) || item.to_label } }
+    attr_ids = attr.to_s
+    attr_ids = (attr_ids.singularize + '_ids').to_sym unless attr_ids.end_with?('_ids')
+    class_name = (html_options["class"] || "" + " without_select2 without_jquery_multiselect").strip!
+
+    items += unauthorized.map do |unauthorized_value|
+      {
+        value: '',
+        label: unauthorized_value.to_s,
+        hidden: true,
+      }
+    end
+
+    if (html_options["data-useds"])
+      JSON.parse(html_options["data-useds"]).map do |disabled_value|
+        items.each do |item|
+          item[:disabled] = true if item[:value] == disabled_value
+        end
+      end
+    end
```

Comment:
I've took the tooltip values from https://github.com/theforeman/foreman/blob/3af4a4d0d3cd8e81b9e2c5d356539b610f52af15/app/assets/javascripts/jquery.multi-select.js#L21
```suggestion
   items.each do |item|
     if options[:mismatches] && options[:mismatches].include?(item[:value])
       item[:disabled] = true
       item[:tooltip] = __("Select this since it belongs to a host")
     elsif options[:useds] && options[:useds].include? ...
       item[:disabled] = true
       item[:tooltip] = __("This is used by a host")
     elsif ....
     end
   end
```